lib/tasks/bus_stop_field.rb

```diff
@@ -0,0 +1,288 @@
+          description: 'Determine the border between the road and the bus stop.'
+        },
+        {
+          name: 'Sidewalk width',
+          column: :sidewalk_width,
+          field_type: :choice,
+          choices: ['More than 36"', 'Less than 36"', 'None'],
+          description: 'Determine if there is a sidewalk at the bus stop location or not. Note the size of the side walk as well.'
+        },
+        {
+          name: 'Bus pull out exists',
+          column: :bus_pull_out_exists,
+          field_type: :boolean,
+          description: 'Determine if there is a formal/constructed pull out in which a bus can pull in and out of.'
+        },
+        {
+          name: 'ADA landing pad',
+          column: :ada_landing_pad,
+          field_type: :boolean,
+          description: "Is there a concrete/brick/asphalt area that is unobstructed and a minimum of 5'x8'?"
+        },
+        {
+          name: 'Obstructions',
+          column: :obstructions,
+          field_type: :choice,
+          choices: [
+            'Yes- tree/branches',
+            'Yes- bollards/structure',
+            'Yes - parking',
+            'Yes - other',
+            'No'
+          ],
+          description: 'Are there any obstructions which make servicing this stop difficult?'
+        }
+      ],
+      'Safety' => [
+        {
+          name: 'Lighting',
+          column: :lighting,
+          field_type: :choice,
+          choices: ["Within 20'", "20' - 50'", 'None'],
+          description: 'Determine if there is lighting in the bus stop.'
+        }
+      ],
+      'Technology' => [
+        {
+          name: 'Solar lighting',
+          column: :solar_lighting,
+          field_type: :boolean,
+          description: 'Determine if this stop has solar lighting.'
+        },
+        {
+          name: 'Has power',
+          column: :has_power,
+          field_type: :choice,
+          choices: ['Yes stub up', 'Yes outlet', 'No'],
+          description: 'Determine if there is power or not.'
+        },
+        {
+          name: 'Real time information',
+          column: :real_time_information,
+          field_type: :choice,
+          choices: ['Yes - solar', 'Yes - power', 'No'],
+          description: 'Determine if this location has a real time information sign.'
+        }
+      ]
+    }
+
+    fields.each_pair do |category, category_fields|
+      category_fields.each.with_index do |category_field, rank|
+        rank += 1
+        field = Field.create! name: category_field.fetch(:name),
+                              rank: rank,
+                              category: category,
+                              description: category_field.fetch(:description),
+                              field_type: category_field.fetch(:field_type),
+                              choices: category_field.fetch(:choices)
+        BusStop.all.each do |stop|
+          BusStopField.create! bus_stop: stop,
```

Comment:
`bus_stop` and `field` are missing in `BusStopField`.

Reply:
oh right. Nested attributes

Reply:
Damn ok I'll try some stuff

Reply:
I don't understand what you're referring to here.

Is there a problem I can help diagnose?

Reply:
Yes. I wrote the migration incorrectly. Now that I've written it correctly (a BusStopField has the attributes `bus_stop` and `field`), the join table is connected by the `bus_stop_id` and `field_id`. I thought that adding a foreign key of `field_name` would solve that problem, but it doesn't.